my/core/query_range.py

```diff
@@ -0,0 +1,526 @@
+"""
+An extension of the my.core.query.select function, allowing you to specify
+a type or key to filter the range by -- this creates a filter function
+given those values, coercing values on the iterable, returning you a
+filtered iterator
+
+See the select_range function below
+"""
+
+import re
+import time
+from functools import lru_cache
+from datetime import datetime, timedelta, date
+from typing import Callable, Iterator, NamedTuple, Optional, Any, Type
+
+import more_itertools
+
+from .query import (
+    QueryException,
+    select,
+    OrderFunc,
+    Where,
+    _handle_generate_order_by,
+    ET,
+)
+
+fromisoformat: Callable[[str], datetime]
```

Comment:
maybe this snippet belongs to `core/compat`? Could remove the duplicate from `common` as well